rs/rosetta-api/src/request_handler.rs

```diff
@@ -633,226 +630,204 @@ impl RosettaRequestHandler {
-                let next = offset
-                    .checked_add(limit)
-                    .ok_or_else(|| ApiError::internal_error("offset + limit overflow"))?;
-                next_offset = Some(i64::try_from(next).map_err(|e| {
-                    ApiError::internal_error(format!("Next offset cannot fit in i64: {}", e))
-                })?);
-            }
-            heights.truncate(limit);
+        if let Some(account) = account_id {
+            command.push_str("AND (from_account = :account_id OR to_account = :account_id OR spender_account = :account_id) ");
+            parameters.push((":account_id", Box::new(account.to_hex())));
         }
 
-        let mut txs: Vec<BlockTransaction> = Vec::new();
-        for i in heights {
-            if i <= last_idx {
-                let hb = blocks.get_hashed_block(&i)?;
-                txs.push(BlockTransaction {
-                    block_identifier: convert::block_id(&hb)?,
-                    transaction: convert::hashed_block_to_rosetta_core_transaction(
-                        &hb,
-                        self.ledger.token_symbol(),
-                    )?,
+        command.push_str("ORDER BY block_idx DESC ");
+
+        command.push_str("LIMIT :limit ");
+        parameters.push((":limit", Box::new(limit)));
+
+        let blocks = block_storage
+            .get_blocks_by_custom_query(
+                command,
+                parameters
+                    .iter()
+                    .map(|(key, param)| {
+                        let param_ref: &dyn rusqlite::ToSql = param.as_ref();
+                        (key.to_owned(), param_ref)
+                    })
+                    .collect::<Vec<_>>()
+                    .as_slice(),
+            )
+            .map_err(|e| ApiError::invalid_block_id(format!("Error fetching blocks: {:?}", e)))?;
+
+        let mut transactions = vec![];
+        for block in blocks.clone().into_iter() {
+            let rosetta_core_block = self.hashed_block_to_rosetta_core_block(block).await?;
+            for transaction in rosetta_core_block.transactions.into_iter() {
+                transactions.push(BlockTransaction {
+                    block_identifier: rosetta_core_block.block_identifier.clone(),
+                    transaction,
                 });
-            } else {
-                return Err(ApiError::InvalidBlockId(true, Default::default()));
             }
         }
+
+        transactions.iter_mut().for_each(|tx| {
+            tx.transaction.operations.iter_mut().for_each(|op| {
+                op.status = Some(STATUS_COMPLETED.to_string());
+            })
+        });
+
+        // Sort the transactions by block index in descending order
+        transactions.sort_by(|a, b| b.block_identifier.index.cmp(&a.block_identifier.index));
+
+        // Is rosetta blocks is empty that means the entire blockchain was traversed but no transactions were found that match the search criteria
```

Comment:
```suggestion
        // If rosetta blocks is empty that means the entire blockchain was traversed but no transactions were found that match the search criteria
```
Isn't this comment mainly related to `transactions.len()`, i.e., it would be a better fit one line later?